runtimevar/etcdvar/etcdvar_test.go

```diff
@@ -23,51 +23,45 @@ import (
 
 	"github.com/google/go-cmp/cmp"
 	"go.etcd.io/etcd/clientv3"
-	"go.etcd.io/etcd/embed"
 	"go.etcd.io/etcd/etcdserver/api/v3rpc/rpctypes"
 	"gocloud.dev/runtimevar"
 	"gocloud.dev/runtimevar/driver"
 	"gocloud.dev/runtimevar/drivertest"
 )
 
+// To run these tests against a local etcd server, first run ./localetcd.sh.
+// Then wait a few seconds for the server to be ready.
+
 var (
-	etcd    *embed.Etcd
-	etcdErr error
+	etcdClient *clientv3.Client
+	etcdError  error
 )
 
 func init() {
-	cfg := embed.NewConfig()
-	cfg.Dir = "default.etcd"
-	etcd, etcdErr = embed.StartEtcd(cfg)
-	if etcdErr != nil {
-		return
-	}
-	select {
-	case <-etcd.Server.ReadyNotify():
+	etcdClient, etcdError = clientv3.NewFromURL("http://localhost:2379")
+	if etcdError == nil {
+		// Check to see if the local etcd is actually running.
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+		defer cancel()
+		_, etcdError = etcdClient.Put(ctx, "unused", "unused")
 	}
 }
 
-type harness struct {
-	client *clientv3.Client
-}
+type harness struct{}
 
 func newHarness(t *testing.T) (drivertest.Harness, error) {
-	if etcdErr != nil {
-		return nil, etcdErr
+	if etcdError != nil {
+		t.Skip("No local Kafka running, see pubsub/kafkapubsub/localkafka.sh")
```

Comment:
etcd?

Also, why not fail? what's the point of running the test if there is no server?

Reply:
Oops, fixed.

We don't run the test if there is no server, we skip the tests.

If we just fail, then the tests on MacOS/Windows would always fail, because the `docker` stuff only works on Linux.